starter-code/main.js

```diff
@@ -6,66 +6,251 @@ const rl = readline.createInterface({ input: process.stdin, output: process.stdo
+						})
+						break;
+
+					case "7":
+						// 7.- List the 10 companies with more employees sorted ascending (show name and employees).
+						db.collection('companies').find({}, { name: 1, _id: 0, number_of_employees: 1 }).sort({ number_of_employees: -1 }).limit(10).toArray((error, result) => {
+							if (error) {
+								console.log(error);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							} else {
+								console.log(result);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							}
+						})
+						break;
+
+					case "8":
+						// 8.- Find the company with the name "Facebook".
+						db.collection('companies').find({ name: "Facebook" }).toArray((error, result) => {
+							if (error) {
+								console.log(error);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							} else {
+								console.log(result);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							}
+						})
+						break;
+
+					case "9":
+						// 9.- How many employees has Facebook?.
+						db.collection('companies').find({ name: "Facebook" }, { name: 1, number_of_employees: 1, _id: 0 }).toArray((error, result) => {
+							if (error) {
+								console.log(error);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							} else {
+								console.log(result);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							}
+						})
+						break;
+
+					case "10":
+						// 10.- List the name of all the products of Facebook.
+						db.collection('companies').find({ name: "Facebook" }, { "products.name": 1, name: 1, _id: 0 }).toArray((error, result) => {
+							if (error) {
+								console.log(error);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							} else {
+								console.log(result);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							}
+						})
+						break;
+
+					case "11":
+						// 11.- List the people that are working at Facebook right now (check relationships field)
+						// This does not work. When I filter by the name, it shows every relationship of the company, and I do not know why
+						db.collection('companies').find({ $and: [{ "relationships": { $elemMatch: { is_past: { $eq: false } } } }] }).toArray((error, result) => {
+							if (error) {
+								console.log(error);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							} else {
+								console.log(result);
+								rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+							}
+						})
+						break;
+
+					case "12":
+						// 12.- List all the companies where "david-ebersman" has worked.
+						db.collection('companies').find({ $and: [{ "relationships": { $elemMatch: { "person.permalink": "david-ebersman" } } }] }, { name: 1, _id: 0 }).toArray((error, result) => {
```

Comment:
Again, posibilidades de simplificación:

`{ $and: [{ 'relationships.person.first_name': 'David', 'relationships.person.last_name': 'Ebersman' }] }`